dissolve-struct-lib/src/main/scala/ch/ethz/dalab/dissolve/optimization/DBCFWSolverTuned.scala

```diff
@@ -972,7 +972,7 @@ class DBCFWSolverTuned[X, Y](
       val rho = 2.0 / (k + 2.0)
       val wAvg = (1.0 - rho) * localModelWeightedAverage.getWeights() + rho * localModel.getWeights()
```

Comment:
this part here is also expensive. here 3 new vector objects are created.. Replacing the addition with += there is also a speedup form 22seconds per round to 16s. Where the test was done using 100k dimensional features.

I think there might be some other places in the code where similar things happen..